static/js/containers/Drawer.js

```diff
@@ -0,0 +1,78 @@
+// @flow
+import React from "react"
+import { connect } from "react-redux"
+import { MDCTemporaryDrawer } from "@material/drawer/dist/mdc.drawer"
+import { withRouter } from "react-router"
+import R from "ramda"
+
+import Navigation from "../components/Navigation"
+
+import { setShowDrawer } from "../actions/ui"
+import { getSubscribedChannels } from "../lib/selectors"
+
+import type { Dispatch } from "redux"
+import type { Channel } from "../flow/discussionTypes"
+import type { Location } from "react-router"
+
+class Drawer extends React.Component {
+  drawerRoot: React$Element<*, *, *>
+  drawer: Object
+
+  props: {
+    location: Location,
+    dispatch: Dispatch,
+    showDrawer: boolean,
+    subscribedChannels: Array<Channel>
+  }
+
+  componentDidMount() {
+    this.drawer = new MDCTemporaryDrawer(this.drawerRoot)
+    this.drawer.listen("MDCTemporaryDrawer:close", this.onDrawerClose)
+  }
+
+  onDrawerClose = () => {
+    const { dispatch, showDrawer } = this.props
+
+    // two MDCTemporaryDrawer:close events seem to fire when the drawer
+    // closes, so this is a little hack to debounce that.
+    if (showDrawer) {
+      dispatch(setShowDrawer(false))
+    }
+  }
+
+  componentWillUnmount() {
+    if (this.drawer) {
+      this.drawer.destroy()
+    }
+  }
+
+  componentDidUpdate() {
+    const { showDrawer } = this.props
+    this.drawer.open = showDrawer
+  }
+
+  render() {
+    const { subscribedChannels, location: { pathname } } = this.props
+
+    return (
+      <aside className="mdc-temporary-drawer mdc-typography" ref={div => (this.drawerRoot = div)}>
+        <nav className="mdc-temporary-drawer__drawer">
+          <header className="mdc-temporary-drawer__header">
+            <div className="mdc-temporary-drawer__header-content">Open Discussions</div>
+          </header>
+          <nav className="mdc-temporary-drawer__content mdc-list">
+            <Navigation subscribedChannels={subscribedChannels} pathname={pathname} />
+          </nav>
+        </nav>
+      </aside>
+    )
+  }
+}
```

Comment:
Like the `Navigation` component, it seems like we can/should just avoid rendering `Drawer` entirely when the visibility is set to false